core/platform/auth/firebase_auth_services_test.py

```diff
@@ -111,32 +113,86 @@ def install(cls, test):
         'list_users',
         'revoke_refresh_tokens',
         'set_custom_user_claims',
-        'update_user',
         'verify_session_cookie',
     ]
 
     def __init__(self):
         self._users = {}
 
-    def _create_user(self, uid=None, **kwargs):
-        """Adds user to storage if new, otherwise raises an error."""
+    def _create_user(self, uid=None, disabled=False):
+        """Adds user to storage if new, otherwise raises an error.
+
+        Args:
+            uid: str. The unique Firebase account ID for the user. The uid
+                argument cannot be made into a positional argument because the
+                Firebase Admin SDK generates a value when it is None. Generating
+                a value is too complicated for tests, however, so we require
+                test writers to provide their own.
+            disabled: bool. Whether the user account is to be disabled.
+
+        Raises:
+            ValueError. The uid argument was not provided.
+            UidAlreadyExistsError. The uid has already been assigned to a user.
+        """
+        if uid is None:
+            raise ValueError('FirebaseAdminSdkStub requires uid to be set')
         if uid in self._users:
             raise firebase_admin.auth.UidAlreadyExistsError(
-                '%s exists' % uid, None, None)
-        # FRAGILE: UserRecord doesn't have a public constructor, so this may
-        # break in future versions of the SDK. OK with taking that risk because
-        # this is only for stubbing purposes.
+                'uid=%r already exists' % uid, None, None)
+        self._set_user(uid, disabled)
+
+    def _update_user(self, uid, disabled=False):
+        """Updates the user in storage if found, otherwise raises an error.
+
+        Args:
+            uid: str. The Firebase account ID of the user.
+            disabled: bool. Whether the user account is to be disabled.
+
+        Raises:
+            UserNotFoundError. The Firebase account has not been created yet.
+        """
+        if uid not in self._users:
+            raise firebase_admin.auth.UserNotFoundError('%s not found' % uid)
+        self._set_user(uid, disabled)
+
+    def _set_user(self, uid, disabled):
+        """Sets the given properties for the corresponding user.
+
+        Args:
+            uid: str. The Firebase account ID of the user.
+            disabled: bool. Whether the user account is to be disabled.
+        """
         self._users[uid] = firebase_admin.auth.UserRecord(
-            firebase_admin.auth.ImportUserRecord(uid, **kwargs).to_dict())
+            # FRAGILE: UserRecord doesn't expose the payload keys as part of its
+            # public API, so this may break in future versions of the SDK. OK
```

Comment:
You might want to add a generic warning at the top of this class, similar to the one I put in test_utils's ElasticSearchStub.

Also I kind of like that this is in the platform test file rather than in test_utils. Maybe I should move ElasticSearchStub and the others there in the future too...